File: game/src/main/java/me/ryleykimmel/brandywine/game/message/listener/ChatMessageListener.java
package me.ryleykimmel.brandywine.game.message.listener;

import me.ryleykimmel.brandywine.game.message.ChatMessage;
import me.ryleykimmel.brandywine.game.model.World;
import me.ryleykimmel.brandywine.game.model.player.Player;
import me.ryleykimmel.brandywine.game.update.blocks.ChatPlayerBlock;

/**
 * Listener for the {@link ChatMessage}.
 */
public final class ChatMessageListener extends GameMessageListener<ChatMessage> {

  /**
   * Constructs a new {@link ChatMessageListener}.
   *
   * @param world The World listening to this {@link ChatMessage}.
   */
  public ChatMessageListener(World world) {
    super(world);
  }

  @Override
  public void handle(Player player, ChatMessage message) {
    player.flagUpdate(ChatPlayerBlock.create(player, message));
  }

}

File: game/src/main/kotlin/plugin/message/ChatMessageListener.kt
package plugin.message

import me.ryleykimmel.brandywine.game.model.player.Player
import me.ryleykimmel.brandywine.game.msg.ChatMessage
import me.ryleykimmel.brandywine.game.update.blocks.ChatPlayerBlock

class ChatMessageListener : MessageListener<Player, ChatMessage> {

    override fun handle(source: Player, message: ChatMessage) {
        source.flagUpdate(ChatPlayerBlock.create(source, message))
    }

}
